src/components/LocationSearch/useNominatimAutocomplete.ts

```diff
@@ -1,35 +1,51 @@
 import React from 'react';
 import debounce from 'lodash/debounce';
 
-const useNominatimAutocomplete = (input: string | unknown[]) => {
-  const [places, setPlaces] = React.useState([]);
+type NominatimResult = {
+  place_id: string;
+  display_name: string;
+  lat: string;
+  lon: string;
+};
+
+export type Place = {
+  id: string;
+  label: string;
+  location: {
+    lat: string;
+    lng: string;
+  };
+};
+
+type Coordinates = {
+  lat: number;
+  lng: number;
+};
+
+const useNominatimAutocomplete = (input: string) => {
+  const [places, setPlaces] = React.useState<Place[]>([]);
 
-  const fetchHandler = async (input) => {
+  const fetchHandler = async (input: string) => {
     try {
-      const fetchUrl = `https://nominatim.openstreetmap.org/search?q=${input}&countrycodes=gb&limit=5&format=json`;
+      const fetchUrl = `https://nominatim.openstreetmap.org/search?q=${encodeURIComponent(
+        input,
+      )}&countrycodes=gb&limit=5&format=json`;
```

Comment:
FYI, `URLSearchParams` does the encoding on your behalf

Reply:
nice!